selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/NewWorkspacePageTest.java

```diff
@@ -0,0 +1,519 @@
+
+    newWorkspace.typeWorkspaceName("");
+
+    newWorkspace.waitErrorMessage("A name is required.");
+    newWorkspace.waitBottomCreateWorkspaceButtonDisabled();
+
+    newWorkspace.typeWorkspaceName("wk");
+
+    newWorkspace.waitErrorMessage("The name has to be more than 3 characters long.");
+    newWorkspace.waitBottomCreateWorkspaceButtonDisabled();
+
+    newWorkspace.typeWorkspaceName("wks");
+
+    newWorkspace.waitErrorMessageDisappearance();
+    newWorkspace.waitBottomCreateWorkspaceButtonEnabled();
+
+    newWorkspace.typeWorkspaceName(NAME_WITH_ONE_HUNDRED_SYMBOLS);
+
+    newWorkspace.waitErrorMessageDisappearance();
+    newWorkspace.waitBottomCreateWorkspaceButtonEnabled();
+
+    newWorkspace.typeWorkspaceName(NAME_WITH_ONE_HUNDRED_SYMBOLS + "p");
+
+    newWorkspace.waitErrorMessage("The name has to be less than 100 characters long.");
+    newWorkspace.waitBottomCreateWorkspaceButtonDisabled();
+
+    newWorkspace.typeWorkspaceName(NAME_WITH_ONE_HUNDRED_SYMBOLS);
+
+    newWorkspace.waitErrorMessageDisappearance();
+    newWorkspace.waitBottomCreateWorkspaceButtonEnabled();
+
+    checkNotValidNames();
+
+    checkValidNames();
+  }
+
+  @Test
+  public void checkStackButtons() {
+    newWorkspace.waitPageLoad();
+    newWorkspace.waitQuickStartButton();
+    newWorkspace.waitStacks(EXPECTED_QUICK_START_STACKS);
+    assertEquals(newWorkspace.getAvailableStacksCount(), EXPECTED_QUICK_START_STACKS_COUNT);
+
+    newWorkspace.clickOnSingleMachineButton();
+    newWorkspace.waitStacks(EXPECTED_SINGLE_MACHINE_STACKS);
+    assertEquals(newWorkspace.getAvailableStacksCount(), EXPECTED_SINGLE_MACHINE_STACKS_COUNT);
+
+    newWorkspace.clickOnAllButton();
+    newWorkspace.waitStacks(EXPECTED_ALL_STACKS);
+    assertEquals(newWorkspace.getAvailableStacksCount(), EXPECTED_ALL_STACKS_COUNT);
+
+    newWorkspace.clickOnMultiMachineButton();
+    newWorkspace.waitStacks(EXPECTED_MULTI_MACHINE_STACKS);
+    assertEquals(newWorkspace.getAvailableStacksCount(), EXPECTED_MULTI_MACHINE_STACKS_COUNT);
+
+    newWorkspace.clickOnAllButton();
+    newWorkspace.waitStacks(EXPECTED_SINGLE_MACHINE_STACKS);
+    newWorkspace.waitStacks(EXPECTED_MULTI_MACHINE_STACKS);
+    assertEquals(
+        newWorkspace.getAvailableStacksCount(),
+        EXPECTED_SINGLE_MACHINE_STACKS_COUNT + EXPECTED_MULTI_MACHINE_STACKS_COUNT);
+
+    newWorkspace.clickOnQuickStartButton();
+    newWorkspace.waitStacksOrder(EXPECTED_QUICK_START_STACKS);
+    newWorkspace.clickNameButton();
+
+    try {
+      newWorkspace.waitStacksOrder(EXPECTED_QUICK_START_STACKS_REVERSE_ORDER);
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/5650", ex);
+    }
+
+    newWorkspace.clickNameButton();
+    newWorkspace.waitStacksOrder(EXPECTED_QUICK_START_STACKS);
+  }
+
+  @Test
+  public void checkFiltersButton() {
+    newWorkspace.waitPageLoad();
```

Comment:
There is a list of verifications many of which look similar, many of which quite different here, and it's hard to understand how they are differ from each other and what is the main logic of test. So, it would be better to have different verifications commented, or extracted into the separate tests.